### PR TITLE
Fix issue with importing Storage Emulator data with nested directory structure

### LINKED ISSUE #4326: Firebase Storage Emulator crashes unexpectedly since v10.3.0
### [REQUIRED] Environment info

**firebase-tools:** 10.4.0

**Platform:** macOS Monterey (12.3)

### [REQUIRED] Test case

- Set up an app with storage emulator and import the existing emulator data (`firebase emulators:start --import ./emulator_data`)

### [REQUIRED] Steps to reproduce

1. Start local emulators including storage emulator (`firebase emulators:start --import ./emulator_data`)2. 
2. Open Emulator UI and go to storage tab

### [REQUIRED] Expected behavior

The storage contents are shown and you can browse through them and upload new files. This works using version `10.2.2` of firebase emulator but since the update it doesn't work anymore.

### [REQUIRED] Actual behavior

The emulator crashes.

```
[debug] [2022-03-18T09:40:32.246Z] Error: ENOENT: no such file or directory, open '/var/folders/pn/tjh55fz97z34qp9h8q6w72bw0000gn/T/firebase/storage/blobs/artistconnect-app.appspot.com%2Fwatermark.wav'
    at openSync (node:fs:585:3)
    at Persistence.readBytes (/Users/matti/.nvm/versions/node/v16.14.0/lib/node_modules/firebase-tools/lib/emulator/storage/persistence.js:37:36)
    at StorageLayer.getBytes (/Users/matti/.nvm/versions/node/v16.14.0/lib/node_modules/firebase-tools/lib/emulator/storage/files.js:98:38)
    at StorageLayer.handleGetObject (/Users/matti/.nvm/versions/node/v16.14.0/lib/node_modules/firebase-tools/lib/emulator/storage/files.js:83:49)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async /Users/matti/.nvm/versions/node/v16.14.0/lib/node_modules/firebase-tools/lib/emulator/storage/apis/firebase.js:74:35
[error] 
[error] Error: An unexpected error has occurred.
```

### PR DESCRIPTION
**Pull request by tohhsinpei**
#### Issue
This PR fixes #4326.

Importing any Storage data exported prior to v10.3.0 causes the emulator to crash.

#### Cause
In #3647 (released as part of v10.3.0), we flattened the emulator directory structure. However, this is not backwards-compatible with previously exported data, which still contains the nested directory structure. In particular, when we copy the imported data, we copy over the old nested directories.

### Scenarios Tested
Tested against [example](https://github.com/charlesfries/firebase-tools-repro) provided in issue and verified that crash does not occur and that we are able to view the file:
repro-fixed

Also wrote two integration tests, one testing import of old nested structure, another testing import of new flattened structure. These will be run as part of `npm run test:storage-emulator-integration`.

### CODE REVIEW

File: scripts/storage-emulator-integration/import/tests.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import * as path from "path";
```
Comment (tonyjhuang) [inline scripts/storage-emulator-integration/import/tests.ts:1]: Just for completeness can you include a test that exports and then imports that data to verify data integrity

File: scripts/storage-emulator-integration/import/tests.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  EMULATORS_SHUTDOWN_DELAY_MS,
+  FIREBASE_EMULATOR_CONFIG,
+  getStorageEmulatorHost,
+  readEmulatorConfig,
+  TEST_SETUP_TIMEOUT,
+} from "../utils";
+
+describe("Import Emulator Data", () => {
+  const FIREBASE_PROJECT = "fake-project-id";
+  const bucket = `${FIREBASE_PROJECT}.appspot.com`;
```
Comment (tonyjhuang) [inline scripts/storage-emulator-integration/import/tests.ts:16]: CONSTANT_CASE?

File: scripts/storage-emulator-integration/import/tests.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  const emulatorConfig = readEmulatorConfig(FIREBASE_EMULATOR_CONFIG);
```
Comment (tonyjhuang) [inline scripts/storage-emulator-integration/import/tests.ts:17]: CONSTANT_CASE?

**Review by tonyjhuang — APPROVED** [review]
Just some comments on testing, thanks!

File: scripts/storage-emulator-integration/import/tests.ts (hunk 2 above)
Comment (tohhsinpei) [reply scripts/storage-emulator-integration/import/tests.ts:16]: Done.

File: scripts/storage-emulator-integration/import/tests.ts (hunk 3 above)
Comment (tohhsinpei) [reply scripts/storage-emulator-integration/import/tests.ts:17]: Done.

File: scripts/storage-emulator-integration/import/tests.ts (hunk 1 above)
Comment (tohhsinpei) [reply scripts/storage-emulator-integration/import/tests.ts:1]: Done.

**Comment by alexjball** [comment]
Thank you for the fix!

**Pull request merged**